benchmark/BenchmarkMeasurementContainer.h

```diff
@@ -0,0 +1,192 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (March of 2023, schlegea@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include <concepts>
+#include <vector>
+#include <memory>
+#include <utility>
+
+#include "util/Timer.h"
+#include "util/json.h"
+#include "../benchmark/BenchmarkMetadata.h"
+
+// All the classes, that hold the measurements of measured benchmarks.
+namespace BenchmarkMeasurementContainer{
+
+// Helper function for adding time entries to the classes.
+/*
+@brief Return execution time of function in seconds.
+
+@tparam Function Best left to type inference.
+
+@param functionToMeasure Must be a function, or callable.
+*/
+template<typename Function>
+  requires std::invocable<Function>
+static float measureTimeOfFunction(const Function& functionToMeasure){
+  ad_utility::timer::Timer
+  benchmarkTimer(ad_utility::timer::Timer::Started);
+  functionToMeasure();
+  benchmarkTimer.stop();
+
+  return ad_utility::timer::Timer::toSeconds(benchmarkTimer.value());
+}
```

Comment:
This still needs the trick from the google benchmark for not optimizing `functionToMeasure`.

Reply:
If you need such tricks, then they have to be inside the `functionToMeasure`. So it is fine to comment that in the examples etc, nothing to do here.

Reply:
I'm pretty sure, that I can't have such tricks as the part of the normal processing, if I want to, for example, stop the processor from optimizing away unused results. There could be benchmarks, where the function has no results, but side effects, which the user wants to measure. For example, the `push_back` function of a new class.
If the class object isn't used somewhere else, the compiler could optimize it away and nothing I could put in `measureTimeOfFunction` could stop it, because I  can't generally know, which things of a benchmark should not be optimized away.
Only the user knows something like that.